client/dashboard/sites/settings-transfer-site/invitation-email-sent.tsx

```diff
@@ -0,0 +1,74 @@
+import { useMutation } from '@tanstack/react-query';
+import { ExternalLink, Spinner } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { sprintf, __ } from '@wordpress/i18n';
+import { useEffect, useState } from 'react';
+import { siteOwnerTransferConfirmMutation } from '../../app/queries';
+import Notice from '../../components/notice';
+import type { SiteTransferResponse } from '../../data/types';
+
+export function InvitationEmailSent( {
+	siteSlug,
+	confirmationHash,
+}: {
+	siteSlug: string;
+	confirmationHash: string;
+} ) {
+	const [ newOwnerEmail, setNewOwnerEmail ] = useState( '' );
+	const [ hasError, setHasError ] = useState( false );
+	const mutation = useMutation( siteOwnerTransferConfirmMutation( siteSlug ) );
+
+	useEffect( () => {
+		mutation.mutate(
+			{ hash: confirmationHash },
+			{
+				onSuccess: ( { email_sent_to }: SiteTransferResponse ) => {
+					setNewOwnerEmail( email_sent_to );
+				},
+				onError: () => {
+					setHasError( true );
+				},
+			}
+		);
+	}, [ confirmationHash ] );
```

Comment:
This doesn't look great, an effect to trigger a mutation is probably a quick way to have bugs. Why aren't we triggering this mutation in an event handler or something?

Reply:
The reason is it's from the confirmation email. The page opens when the user click the button in the email, I don't think there is other places to trigger the mutation 🥲